### PR TITLE
Port LoRA to multi-backend

### PR DESCRIPTION
**Pull request by abheesht17**
Resolves https://github.com/keras-team/keras-nlp/issues/1141

Yet to figure out how to track GPU memory.

### CODE REVIEW

File: examples/nlp/parameter_efficient_finetuning_of_gpt2_with_lora.py [hunk 1]
```diff
@@ -37,26 +37,42 @@
 ## Setup
 
 Before we start implementing the pipeline, let's install and import all the
-libraries we need. We'll be using the KerasNLP library.
-
-Secondly, let's enable mixed precision training. This will help us reduce the
-training time.
+libraries we need. We'll be using the KerasNLP library. Also, let's choose
```
Comment (mattdangerw) [inline examples/nlp/parameter_efficient_finetuning_of_gpt2_with_lora.py:40]: Maybe copy some language from [here](https://github.com/keras-team/keras-io/pull/1432/files#r1275615099). In particular a link to Keras Core would be good.

File: examples/nlp/parameter_efficient_finetuning_of_gpt2_with_lora.py [hunk 2]
```diff
@@ -37,26 +37,42 @@
...
 """
 
 """shell
-pip install keras-nlp -q
+pip install keras-nlp nvidia-ml-py3 -q
 """
 
+import os
+
+os.environ["KERAS_BACKEND"] = "jax"
```
Comment (mattdangerw) [inline examples/nlp/parameter_efficient_finetuning_of_gpt2_with_lora.py:51]: `# or "tensorflow" or "torch"`

File: examples/nlp/parameter_efficient_finetuning_of_gpt2_with_lora.py [hunk 3]
```diff
@@ -37,26 +37,42 @@
...
-policy = keras.mixed_precision.Policy("mixed_float16")
-keras.mixed_precision.set_global_policy(policy)
+
+"""
+Before we move on, one thing we should do is "turn on the memory growth" so that
+memory is dynamically allocated to processes at runtime.
+This will allow us to make a fairer comparison between GPT-2 and LoRA.
+"""
+
+gpu = tf.config.list_physical_devices("GPU")[0]
```
Comment (mattdangerw) [inline examples/nlp/parameter_efficient_finetuning_of_gpt2_with_lora.py:70]: I think we can kill this code/text block, and do the following above...

```python
os.environ["KERAS_BACKEND"] = "jax"  # or "tensorflow" or "torch"
# Prevent tensorflow from allocating all available memory so we can
# monitor memory usage growth in this guide.
os.environ["TF_FORCE_GPU_ALLOW_GROWTH"] = "true"
```

File: examples/nlp/parameter_efficient_finetuning_of_gpt2_with_lora.py [hunk 4]
```diff
@@ -427,7 +449,7 @@ def call(self, inputs):
 del loss
 
 # This resets "peak" memory usage to "current" memory usage.
-tf.config.experimental.reset_memory_stats("GPU:0")
+# tf.config.experimental.reset_memory_stats("GPU:0")
```
Comment (mattdangerw) [inline examples/nlp/parameter_efficient_finetuning_of_gpt2_with_lora.py:452]: Just delete this line? Not needed now that we are using nvidia directly to monitor GPU memory.

**Review by mattdangerw — APPROVED** [review]
Looks great! Few misc comments.

**Comment by abheesht17** [comment]
Not yet ready for merge

**Comment by abheesht17** [comment]
Commenting to prevent the bot from closing this PR.

**Pull request closed without merging**